draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -356,869 +524,832 @@ document are to be interpreted as described in {{RFC2119}}.
-  (A=-3, B=41058363725152142129326129780047268409114441015993725554835256314039467401291).
+The following table lists recommended algorithms for different curves and
+mappings. To select a suitable algorithm, choose the mapping associated with
+the target curve. For example, Elligator2 is the recommended encoding for
+Curve25519, whereas Simple SWU is the recommended encoding for P-256.
+When the required hashing requires to be used in a protocol proven in the
+random oracle model, applications SHOULD use the Random Oracle construction
+given in {{rom}}.
 
-H2C-P384-SHA512-Icart- is defined as follows:
+A suite is a bundle of algorithms that provides concrete recommendations for
+hashing bitstrings into points of specific elliptic curve groups. Each suite is
+a tuple (E, H, f, ROM) such that
 
-* The destination group is the set of points on the NIST P-384 elliptic curve, with
-  curve parameters as specified in {{DSS}} (Section D.1.2.4) and
-  {{RFC5114}} (Section 2.7).
-* hash2base is defined as {#hashtobase} with the hash function defined as
-  SHA-512 as specified in {{RFC6234}}, and p set to the prime field used in
-  P-384 (2^384 - 2^128 - 2^96 + 2^32 - 1).
-* HashToCurve is defined to be {#icart} with A and B taken from the definition of P-384
-  (A=-3, B=27580193559959705877849011840389048093056905856361568521428707301988689241309860865136260764883745107765439761230575).
+* E, is the elliptic curve group.
+* H, is the cryptographic hash function used by hash2base.
+* f, is an encoding function compatible with E.
+* ROM, is a boolean flag indicating whether or not to use the random oracle construction.
 
-H2C-Curve25519-SHA512-Elligator2-Clear is defined as follows:
+This document describes the following set of ciphersuites
 
-* The destination group is the points on Curve25519, with
-  curve parameters as specified in {{RFC7748}} (Section 4.1).
-* hash2base is defined as {#hashtobase} with the hash function defined as
-  SHA-512 as specified in {{RFC6234}}, and p set to the prime field used in
-  Curve25519 (2^255 - 19).
-* HashToCurve is defined to be {#elligator2} with the curve function defined
-  to be the Montgomery form of Curve25519 (y^2 = x^3 + 486662x^2 + x) and
-  N = 2.
-* The final output is multiplied by the cofactor of Curve25519, 8.
+| Suite ID | E | H | f | ROM |
+|----------|---|---|---|-----|
+| H2C-0000 | P256         |  SHA256 | Simple SWU | True |
+| H2C-0000 | P384         |  SHA512 | Icart      | True |
+| H2C-0000 | curve25519   |  SHA512 | Elligator2 | True |
+| H2C-0000 | curve448     |  SHA512 | Elligator2 | True |
+| H2C-0000 | edwards25519 |  SHA512 | Elligator2 | True |
+| H2C-0000 | edwards448   |  SHA512 | Elligator2 | True |
+| H2C-0000 | SECP256K1    |  SHA512 | FT         | True |
+| H2C-0000 | BLS12381     |  SHA512 | FT         | True |
 
-H2C-Curve448-SHA512-Elligator2-Clear is defined as follows:
+## Cofactor multiplication
```

Comment:
It seems odd that callers of `hash2curve` may have to *also* multiply by the cofactor to get a usable point in the curve. I think we should bake this step into the algorithms, i.e., always include a step that multiplies by the cofactor in each algorithm. This works for all curves, since for curves where p=n (h=1) nothing changes.

Reply:
Now it is included a cofactor multiplication for each encoding in the last step.